opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectMappableContent.tsx

```diff
@@ -0,0 +1,270 @@
+import { FormikConfig } from 'formik/dist/types';
+import { ExternalReferencesValues } from '@components/common/form/ExternalReferencesField';
+import { StixCoreObjectMappableContentFieldPatchMutation } from '@components/common/stix_core_objects/__generated__/StixCoreObjectMappableContentFieldPatchMutation.graphql';
+import MarkdownField from '../../../../components/fields/MarkdownField';
+import { SubscriptionFocus } from '../../../../components/Subscription';
+import RichTextField from '../../../../components/fields/RichTextField';
+import { fieldSpacingContainerStyle } from '../../../../utils/field';
+import { useFormatter } from '../../../../components/i18n';
+import { useIsEnforceReference, useSchemaEditionValidation } from '../../../../utils/hooks/useEntitySettings';
+import useApiMutation from '../../../../utils/hooks/useApiMutation';
+
+export const stixCoreObjectMappableContentFieldPatchMutation = graphql`
+  mutation StixCoreObjectMappableContentFieldPatchMutation(
+    $id: ID!
+    $input: [EditInput]!
+    $commitMessage: String
+    $references: [String]
+  ) {
+    stixDomainObjectEdit(id: $id) {
+      fieldPatch(input: $input, commitMessage: $commitMessage, references: $references) {
+        ...StixCoreObjectContent_stixCoreObject
+      }
+    }
+  }
+`;
+
+interface StixCoreObjectMappableContentProps {
+  containerData: StixCoreObjectContent_stixCoreObject$data;
+  handleDownloadPdf?: () => void;
+  handleTextSelection?: (t: string) => void;
+  askAi: boolean;
+  editionMode: boolean;
+  mappedStrings?: string[];
+  suggestedMappedStrings?: string[];
+}
+
+interface StixCoreObjectMappableContentValues {
+  content: string
+  description: string
+  message?: string | null
+  references?: ExternalReferencesValues
+}
+
+const StixCoreObjectMappableContent: FunctionComponent<StixCoreObjectMappableContentProps> = ({
+  containerData,
+  handleDownloadPdf,
+  handleTextSelection,
+  askAi,
+  editionMode,
+  mappedStrings = [],
+  suggestedMappedStrings = [],
+}) => {
+  const { t_i18n } = useFormatter();
+  let { description, contentField } = containerData;
+  const [selectedTab, setSelectedTab] = useState(editionMode ? 'write' : 'preview');
+  const basicShape = {
+    content: Yup.string().nullable(),
+    description: Yup.string().nullable(),
+  };
+  let validator = null;
+  if (containerData.entity_type === 'Report') {
+    validator = useSchemaEditionValidation('Report', basicShape);
+  } else if (containerData.entity_type === 'Grouping') {
+    validator = useSchemaEditionValidation('Grouping', basicShape);
+  } else if (containerData.entity_type === 'Case-Incident') {
+    validator = useSchemaEditionValidation('Case-Incident', basicShape);
+  } else if (containerData.entity_type === 'Case-Rfi') {
+    validator = useSchemaEditionValidation('Case-Rfi', basicShape);
+  } else if (containerData.entity_type === 'Case-Rft') {
+    validator = useSchemaEditionValidation('Case-Rft', basicShape);
```

Comment:
looks like it could be simplified with :
`validator = useSchemaEditionValidation(containerData.entity_type, basicShape);`